ibuffer-projectile.el

```diff
@@ -98,10 +98,16 @@ If the file is not in a project, then nil is returned instead."
 (defun ibuffer-projectile-generate-filter-groups ()
   "Create a set of ibuffer filter groups based on the projectile root dirs of buffers."
   (let ((roots (ibuffer-remove-duplicates
-                (delq nil (mapcar 'ibuffer-projectile-root (buffer-list))))))
+                (delq nil (mapcar (lambda (buf)
+                                    (let ((root (ibuffer-projectile-root buf)))
+                                      (when root
+                                        (cons (with-current-buffer buf
```

Comment:
Rather than splice extra projectile information into the items here, please simply modify `ibuffer-projectile-root` to return a cons cell for each entry. (`ibuffer-vc` uses the same approach.)